lighthouse-core/scripts/i18n/collection-util.js

```diff
@@ -0,0 +1,281 @@
+        example: example,
+      };
+    }
+  }
+
+  // add examples for direct ICU replacement
+  idx = 0;
+  // eslint-disable-next-line guard-for-in
+  for (const [key, value] of Object.entries(examples)) {
+    if (!newMessage.includes(`{${key}}`)) continue;
+    const eName = `ICU_${idx++}`;
+    newMessage = newMessage.replace(`{${key}}`, `$${eName}$`);
+
+    placeholders[eName] = {
+      content: `{${key}}`,
+      example: value,
+    };
+  }
+
+  return {message: newMessage, placeholders};
+}
+
+
+/**
+ * @param {Record<string, ICUMessageDefn>} strings
+ */
+function createPsuedoLocaleStrings(strings) {
+  /** @type {Record<string, ICUMessageDefn>} */
+  const psuedoLocalizedStrings = {};
+  for (const [key, defn] of Object.entries(strings)) {
+    const message = defn.message;
+    const psuedoLocalizedString = [];
+    let braceCount = 0;
+    let inPlaceholder = false;
+    let useHatForAccentMark = true;
+    for (const char of message) {
+      psuedoLocalizedString.push(char);
+      if (char === '$') {
+        inPlaceholder = !inPlaceholder;
+        continue;
+      }
+      if (inPlaceholder) {
+        continue;
+      }
+
+      if (char === '{') {
+        braceCount++;
+      } else if (char === '}') {
+        braceCount--;
+      }
+
+      // Hack to not change {plural{ICU}braces} nested an odd number of times.
+      // ex: "{itemCount, plural, =1 {1 link found} other {# links found}}"
+      // becomes "{itemCount, plural, =1 {1 l̂ín̂ḱ f̂óûńd̂} other {# ĺîńk̂ś f̂óûńd̂}}"
+      // ex: "{itemCount, plural, =1 {1 link {nested_replacement} found} other {# links {nested_replacement} found}}"
+      // becomes: "{itemCount, plural, =1 {1 l̂ín̂ḱ {nested_replacement} f̂óûńd̂} other {# ĺîńk̂ś {nested_replacement} f̂óûńd̂}}"
+      if (braceCount % 2 === 1) continue;
+
+      // Add diacritical marks to the preceding letter, alternating between a hat ( ̂ ) and an acute (´).
+      if (/[a-z]/i.test(char)) {
+        psuedoLocalizedString.push(useHatForAccentMark ? `\u0302` : `\u0301`);
+        useHatForAccentMark = !useHatForAccentMark;
+      }
+    }
+    psuedoLocalizedStrings[key] = {
+      message: psuedoLocalizedString.join(''),
+      placeholders: defn.placeholders,
+    };
+  }
+  return psuedoLocalizedStrings;
+}
+
+/**
+ * @param {Record<string, ICUMessageDefn>} messages
+ * @returns {Record<string, ICUMessageDefn>}
+ */
+function bakePlaceholders(messages) {
+  for (const key in messages) {
+    if (!Object.prototype.hasOwnProperty.call(messages, key)) continue;
+
+    delete messages[key]['description'];
+
+    let message = messages[key]['message'];
+    const placeholders = messages[key]['placeholders'];
+
+    for (const placeholder in placeholders) {
+      if (!Object.prototype.hasOwnProperty.call(placeholders, placeholder)) continue;
+
+      const content = placeholders[placeholder]['content'];
+      const re = new RegExp('\\$' + placeholder + '\\$');
```

Comment:
can this be done with just strings? seems like we're not doing anything fancy with globals/parts and escaping all the special characters anyhow